Provider/src/main/java/com/spotify/confidence/Confidence.kt

```diff
@@ -88,6 +86,38 @@ class Confidence internal constructor(
     ) {
         eventSenderEngine.emit(definition, payload, getContext())
     }
+
+    override fun track(eventProducer: EventProducer) {
+        coroutineScope.launch {
+            eventProducer
+                .events()
+                .collect { event ->
+                    eventSenderEngine.emit(
+                        definition = event.name,
+                        event.message,
+                        getContext()
+                    )
+                }
+        }
+
+        coroutineScope.launch {
+            eventProducer.contextChanges()
+                .collect(this@Confidence::putContext)
+        }
+        eventProducers.add(eventProducer)
+    }
+
+    override fun stop() {
+        for (producer in eventProducers) {
+            producer.stop()
+        }
+        if (parent == null) {
+            eventSenderEngine.stop()
+        } else {
+            // no-op for child confidence
+            coroutineScope.cancel()
```

Comment:
shouldn't this also happen for `parent==null`?

Reply:
yes, you are right.